core/tx_pool.go

```diff
@@ -1254,6 +1282,7 @@ func (as *accountSet) flatten() []common.Address {
 type txLookup struct {
 	all  map[common.Hash]*types.Transaction
 	lock sync.RWMutex
```

Comment:
Do we still need this mutex @fjl ? With the singleton add rework, I'm not sure about this any more.